WordPress/gutenberg
# Site Editor: Add theme.json to export file

## Description
Resolves #27528
Supersedes #36619

Adds global style settings (as theme.json) to the site export file.

PR also overrides the core REST API endpoint for export functionality. 

## Testing Instructions
1. Go to Appearance > Editor
2. Open the "More tools and Options" dropdown and export.
3. Confirm that the archive contains the theme.json
4. Confirm that user-applied changes are correctly exported.

## Types of changes
Enhancement

## Checklist:
